src/wrapper/objects/auto_byte_array.rs

```diff
@@ -58,17 +58,45 @@ impl<'a, 'b> AutoByteArray<'a, 'b> {
         self.ptr.as_ptr()
     }
 
-    /// Commits the result of the array, if it is a copy
+    fn commit_byte_array_elements(&mut self) -> Result<()> {
+        jni_void_call!(
+            self.env.get_native_interface(),
+            ReleaseByteArrayElements,
+            *self.obj,
+            self.ptr.as_mut(),
+            sys::JNI_COMMIT
+        );
+        Ok(())
+    }
+
+    fn release_byte_array_elements(&mut self) -> Result<()> {
+        jni_void_call!(
+            self.env.get_native_interface(),
+            ReleaseByteArrayElements,
+            *self.obj,
+            self.ptr.as_mut(),
+            self.mode as i32
+        );
+        Ok(())
+    }
+
+    /// Commits the changes to the array, if it is a copy
     pub fn commit(&mut self) {
-        let res = self
-            .env
-            .commit_byte_array_elements(*self.obj, unsafe { self.ptr.as_mut() });
+        let res = self.commit_byte_array_elements();
         match res {
             Ok(()) => {}
             Err(e) => debug!("error committing byte array: {:#?}", e),
         }
     }
 
+    /// Don't commit the changes to the array on release (if it is a copy).
+    /// This has no effect if the array is not a copy.
```

Comment:
I wonder if it shall be a precondition that a user shall check? Whether an array is a copy is out of a user's control. Therefore, they can only rely on `discard` if they ensure that an array is a copy.

What would be the best approach here? A requirement that a user checks if an array is a copy, and an error here? Or panic? Or keep as is?

Reply:
I think it's the user's responsibility; user should have checked the array is a copy or not long before calling this method, and there's a public method for doing that.

The most appropriate thing would be a warning, but I don't want to potentially pollute the output with it. Besides, the documentation is clear; I would leave it as it is.